### PR TITLE
[vcpkg] Rename `port_versions` to `versions`

### PR DESCRIPTION
**Pull request by strega-nil**
After bikeshedding, we have decided upon a different name for the existing `port_versions` directory and we want to rename it before people start depending on it.

### CODE REVIEW

File: docs/specifications/versioning.md [hunk 1]
```diff
@@ -319,9 +319,9 @@ Although the concept of package versions has always been present in vcpkg, the c
 
 With the introduction of versioning constraints, it is now possible that a package depends on a port version that does not match the one available locally.  This raises a problem as vcpkg needs to know how to acquire the port files for the requested version.
 
-To solve this problem, a new set of metadata needs to be introduced. This specification proposes a that a new “port_versions” folder is added as part of a registry. In the main vcpkg registry, this means a new root level port_versions directory. 
+To solve this problem, a new set of metadata needs to be introduced. This specification proposes a that a new "versions" folder is added as part of a registry. In the main vcpkg registry, this means a new root level versions directory.
```
Comment (BillyONeal) [inline docs/specifications/versioning.md:322]: directory, not folder. (Sorry, I know you didn't edit, but still)

**Review by BillyONeal — APPROVED** [review]

File: docs/specifications/versioning.md (hunk 1 above)
Comment (strega-nil) [reply docs/specifications/versioning.md:322]: prolly not gonna make that change since I'm just doing the boring rename thing.

Comment (BillyONeal) [reply docs/specifications/versioning.md:322]: Only worry about it if you need to spin this again. (I view this comment as no different than you asking me to sort cmake vars in the vcpkg_minimum_required PR :) )

**Review by vicroms — APPROVED** [review]

**Comment by strega-nil** [comment]
Thanks @vicroms

**Comment by strega-nil** [comment]
Thanks @vicroms

File: docs/specifications/versioning.md (hunk 1 above)
Comment (Sjoeborg) [reply docs/specifications/versioning.md:322]: whats the difference really?

**Pull request merged**